python/tests/test_highlevel.py

```diff
@@ -885,6 +885,64 @@ def test_virtual_root_arg(self):
         assert self.t.mrca(0, 5) == 5
 
 
+class TestPathLength:
+    t = tskit.Tree.generate_balanced(9)
+    #         16
+    #    ┏━━━━┻━━━┓
+    #    ┃       15
+    #    ┃     ┏━━┻━┓
+    #   11     ┃   14
+    #  ┏━┻━┓   ┃  ┏━┻┓
+    #  9  10  12  ┃ 13
+    # ┏┻┓ ┏┻┓ ┏┻┓ ┃ ┏┻┓
+    # 0 1 2 3 4 5 6 7 8
+
+    def test_tmrca_leaf(self):
+        assert self.t.path_length(0, 16) == 3
+        assert self.t.path_length(16, 0) == 3
+        assert self.t.path_length(7, 16) == 4
+
+    def test_equal_depth(self):
+        assert self.t.path_length(5, 16) == self.t.depth(5)
+
+    def test_two_leaves(self):
+        assert self.t.path_length(0, 8) == 7
+
+    def test_two_leaves_depth(self):
+        assert self.t.path_length(0, 8) == self.t.depth(0) + self.t.depth(8)
+
+    # def test_less_than_two_args(self):
+    #     with pytest.raises(ValueError):
+    #         self.t.path_length(1)
+    #
+    # def test_no_args(self):
```

Comment:
I guess we should test the interface a little all right, so how about something like
```python
@pytest.mark.parametrize("args", [[], [1], [1, 2, 3]])
def test_bad_num_args(self, args):
     with pytest.raises(TypeError):
            self.t.path_length(*args)

@pytest.mark.parametrize("bad_arg", [[], "1"])
def test_bad_arg_type(self, bad_arg)
      with pytest.raises(ValueError):
             self.t.path_length(0, bad_arg)
       with pytest.raises(ValueError):
             self.t.path_length(bad_arg, 0) 
```
I'm not sure if they second one will always raise a ValueError or if some things are TypeErrors. See what works, we don't need to worry too much about testing all possibilities here.

Reply:
Indeed it worked with `TypeError`.
I followed the tutorial on squashing, very useful! I forgot to edit the text of the commit message afterwards, I don't know if it's a problem.

Just one little thing, in the tutorial on squashing it's `git rebase -i upstream/master` where it should be `git rebase -i upstream/main`. Took me a bit of time to get the origin of the error git was raising when I used the first command.

Reply:
Ah, good to know, thanks.